From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: auvred <aauvred@gmail.com>
Date: Wed, 26 Mar 2025 15:11:27 +0300
Subject: [PATCH] Create one checker per CPU


diff --git a/internal/compiler/program.go b/internal/compiler/program.go
index 837b41b21..bf2e73cb0 100644
--- a/internal/compiler/program.go
+++ b/internal/compiler/program.go
@@ -2,6 +2,7 @@ package compiler
 
 import (
 	"fmt"
+	"runtime"
 	"slices"
 	"sync"
 
@@ -214,7 +215,7 @@ func (p *Program) CheckSourceFiles() {
 
 func (p *Program) createCheckers() {
 	p.checkersOnce.Do(func() {
-		p.checkers = make([]*checker.Checker, core.IfElse(p.programOptions.SingleThreaded, 1, 4))
+		p.checkers = make([]*checker.Checker, core.IfElse(p.programOptions.SingleThreaded, 1, runtime.GOMAXPROCS(0)))
 		wg := core.NewWorkGroup(p.programOptions.SingleThreaded)
 		for i := range p.checkers {
 			wg.Queue(func() {
