From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: auvred <aauvred@gmail.com>
Date: Mon, 7 Apr 2025 16:56:15 +0300
Subject: [PATCH] Do not sort symbols in getNamedMembers


diff --git a/internal/checker/checker.go b/internal/checker/checker.go
index 01bc42ae5..4961bb135 100644
--- a/internal/checker/checker.go
+++ b/internal/checker/checker.go
@@ -20401,7 +20401,6 @@ func (c *Checker) getNamedMembers(members ast.SymbolTable) []*ast.Symbol {
 			result = append(result, symbol)
 		}
 	}
-	c.sortSymbols(result)
 	return result
 }
 
