# prefer-ts-expect-error

## Rule Details

Prefer `@ts-expect-error` over `@ts-ignore` in TypeScript directive comments.

Examples of **incorrect** code for this rule:

```typescript
// @ts-ignore
```

Examples of **correct** code for this rule:

```typescript
// @ts-expect-error
```

## Original Documentation

https://typescript-eslint.io/rules/prefer-ts-expect-error
